app/controllers/switch-psp/verify-psp-integration.controller.js

```diff
@@ -34,7 +39,7 @@ async function startPaymentJourney (req, res, next) {
       payment_provider: targetCredential.payment_provider,
       description: 'Live payment to verify new PSP',
       reference: 'VERIFY_PSP_INTEGRATION',
-      return_url: urljoin(req.headers && req.headers.origin, formatAccountPathsFor(paths.account.switchPSP.receiveVerifyPSPIntegrationPayment, req.account.external_id))
+      return_url: urljoin(req.headers && getHost(req.headers.referer), formatAccountPathsFor(paths.account.switchPSP.receiveVerifyPSPIntegrationPayment, req.account.external_id))
```

Comment:
Another option we might consider here is using the self service base URL which should be templated into the environment, that would avoid any ambiguity with browser header support/ extensions messing or spoofing headers